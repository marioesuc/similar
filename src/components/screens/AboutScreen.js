// Component that renders screen related to the video viewer

import React, { Component } from 'react';
import { View, Image, Text, StyleSheet } from 'react-native';
import LinearGradient from 'react-native-linear-gradient';
import * as Colors from './styles/Colors';

class About extends Component {

  render() {
    return (

      <LinearGradient 
        colors={[Colors.backgroundTopColor, Colors.backgroundBottomColor]}
        style={styles.container}
      >
        <View style={styles.logoContainer}>
          <Image style={styles.logo} source={require('../../../images/similar_logo.png')} />
        </View>

          <View style={styles.textBox}>
            <Text style={styles.titleText}>¿Qué es Similar?</Text>
            <Text style={styles.titleDesc}>Similar no es una aplicación cualquiera para aprender idiomas. Nosotros, los creadores, creemos que un buen método de aprendizaje es fundamental para memorizar de la forma más eficiente y rápida posible.</Text>
            <Text style={styles.titleDesc}>Para ello combinamos el uso de las famosas tarjetas de aprendizaje (muy utilizadas en el estudio del idioma Chino) con el reaprovechamiento de los recursos que ofrece el nexo común del latín entre nuestro idioma, el español, y el inglés.</Text>
          </View>

      </LinearGradient>
    );
  }
}

const styles = StyleSheet.create({
  container: {
    flex: 1,
    backgroundColor: Colors.appBackground,
    alignItems: 'center',
    padding: 20
  },
  logo: {
    width: 200,
    height: 173
  },
  viewer: { height: 200 },
  textBox: {
    backgroundColor: 'white',
    width: '95%',
    margin: 10,
    padding: 10,
    elevation: 2,
    borderWidth: 0
  },
  titleText: {
    fontSize: 20,
    fontWeight: 'bold'
  },
  titleDesc: {
    padding: 10
  }
});

export default About;
