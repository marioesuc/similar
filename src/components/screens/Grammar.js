// Component that renders screen related to the grammar videos

import React, { Component } from 'react';
import { View, FlatList, StyleSheet } from 'react-native';
import ListItem from '../common/ListItem';
import * as Colors from './styles/Colors';

class Grammar extends Component {

  render() {
    return (

      <View style={styles.container}>
        <FlatList
          data={[
            { picUrl: 'https://img.youtube.com/vi/3wmVgKjMpC8/0.jpg', title: 'Título del vídeo' },
            { picUrl: 'https://img.youtube.com/vi/3wmVgKjMpC8/0.jpg', title: 'Título del vídeo' },
            { picUrl: 'https://img.youtube.com/vi/3wmVgKjMpC8/0.jpg', title: 'Título del vídeo' },
            { picUrl: 'https://img.youtube.com/vi/3wmVgKjMpC8/0.jpg', title: 'Título del vídeo' },
            { picUrl: 'https://img.youtube.com/vi/3wmVgKjMpC8/0.jpg', title: 'Título del vídeo' },
            { picUrl: 'https://img.youtube.com/vi/3wmVgKjMpC8/0.jpg', title: 'Título del vídeo' },
            { picUrl: 'https://img.youtube.com/vi/3wmVgKjMpC8/0.jpg', title: 'Título del vídeo' },
            { picUrl: 'https://img.youtube.com/vi/3wmVgKjMpC8/0.jpg', title: 'Título del vídeo' },
            { picUrl: 'https://img.youtube.com/vi/3wmVgKjMpC8/0.jpg', title: 'Título del vídeo' },
            { picUrl: 'https://img.youtube.com/vi/3wmVgKjMpC8/0.jpg', title: 'Título del vídeo' },
            { picUrl: 'https://img.youtube.com/vi/3wmVgKjMpC8/0.jpg', title: 'Título del vídeo' }
          ]}
          renderItem={({ item }) => <ListItem picUrl={item.picUrl} text={item.title} />}
          keyExtractor={(item, index) => index.toString()}
        />
      </View>
    );
  }
}

const styles = StyleSheet.create({
  container: {
    flex: 1,
    backgroundColor: Colors.appBackground,
  }
});

export default Grammar;
