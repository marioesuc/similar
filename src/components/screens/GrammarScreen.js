// Component that renders screen related to the grammar videos

import React, { Component } from 'react';
import { FlatList, StyleSheet } from 'react-native';
import { connect } from 'react-redux';
import LinearGradient from 'react-native-linear-gradient';

import ListItem from '../common/ListItem';
import * as Colors from './styles/Colors';
import { loadGrammarData } from '../../actions';

class Grammar extends Component {

  componentWillMount() {
    this.props.loadGrammarData();
  }

  render() {
    return (

      <LinearGradient 
        colors={[Colors.backgroundTopColor, Colors.backgroundBottomColor]}
        style={styles.container}
      >
        <FlatList
          data={this.props.grammarData}
          renderItem={({ item }) => <ListItem picUrl={item.picUrl} videoUrl={item.videoUrl} title={item.title} desc={item.desc} />}
          keyExtractor={(item, index) => index.toString()}
        />
      </LinearGradient>
    );
  }
}

const styles = StyleSheet.create({
  container: {
    flex: 1,
    backgroundColor: Colors.appBackground,
  }
});

const mapStateToProps = ({ grammar }) => {
  const { grammarData } = grammar;
  return { grammarData };
};

export default connect(mapStateToProps, { loadGrammarData })(Grammar);
