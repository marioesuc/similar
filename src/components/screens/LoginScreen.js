// Component that renders screen related to the login

import React, { Component } from 'react';
import { TextInput, View, Image, Text, StyleSheet, TouchableOpacity } from 'react-native';
import { Actions } from 'react-native-router-flux';
import { connect } from 'react-redux';
import LinearGradient from 'react-native-linear-gradient';
import { loginEmailChanged, loginPasswordChanged, loginUser } from '../../actions';
import * as Colors from './styles/Colors';

class Login extends Component {

  onEmailChange(text) {
    //loginEmailChanged es la función Action Creator que hemos importado arriba
    this.props.loginEmailChanged(text);
  }

  onPasswordChange(text) {
    this.props.loginPasswordChanged(text);
  }

  onButtonPress() {
    const { email, password } = this.props;

    this.props.loginUser({ email, password });
  }

  render() {
    return (
      <LinearGradient 
        colors={[Colors.backgroundTopColor, Colors.backgroundBottomColor]}
        style={styles.container}
      >
        <View style={styles.logoContainer}>
          <Image style={styles.logo} source={require('../../../images/similar_logo.png')} />
        </View>
        <TextInput
          value={this.props.email}
          onChangeText={this.onEmailChange.bind(this)}
          placeholder={'E-mail'}
          style={styles.input}
        />
        <TextInput
          value={this.props.password}
          onChangeText={this.onPasswordChange.bind(this)}
          placeholder={'Contraseña'}
          secureTextEntry
          style={styles.input}
        />
        
        <TouchableOpacity
          style={styles.buttonContainer}
          onPress={this.onButtonPress.bind(this)}
        >
          <Text style={styles.buttonText}>ENTRAR</Text>
        </TouchableOpacity>

        <Text style={styles.errorTextStyle}>
          {this.props.error}
        </Text>
        <TouchableOpacity style={styles.buttonBottom}
        onPress={() => {
                    Actions.register();
                  }}
        >
          <Text
            style={styles.bottomButtonText}
            title="Registro"
          >
            Registro
          </Text>
        </TouchableOpacity>
      </LinearGradient>
    );
  }
}

const styles = StyleSheet.create({
  container: {
    flex: 1,
    alignItems: 'center',
    justifyContent: 'center',
    backgroundColor: Colors.appBackground,
  },
  input: {
    width: 300,
    height: 44,
    padding: 10,
    marginBottom: 10,
  },
  logo: {
    width: 250,
    height: 200
  },
  button: {
    padding: 20
  },
  errorTextStyle: {
    marginTop: 20,
    fontSize: 20,
    alignSelf: 'center',
    color: Colors.errorText
  },
  buttonsContainer: {
    flexDirection: 'row',
    width: 300
  },
  buttonContainer: {
    backgroundColor: Colors.transparentButton,
    paddingVertical: 15,
    width: 300
  },
  buttonText: {
    textAlign: 'center',
    color: Colors.transparentButtonText,
    fontWeight: '700'
  },
  bottomButtonText: {
    textAlign: 'center',
    color: '#FFF',
    fontWeight: '700'
  },
  buttonsSeparation: {
    flex: 1,
    marginRight: 10
  },
  registerButton: { flex: 1 },
  buttonBottom: {
    backgroundColor: Colors.registerButton,
    paddingVertical: 15,
    width: '100%',
    position: 'absolute',
    bottom: 0
  }
});

const mapStateToProps = ({ auth }) => {
  const { email, password, error, loading } = auth;
  return { email, password, error, loading };
};


export default connect(mapStateToProps, { 
  loginEmailChanged, loginPasswordChanged, loginUser 
})(Login);
