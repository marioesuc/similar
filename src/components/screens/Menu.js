// Component that renders screen related to main menu options

import React, { Component } from 'react';
import { Alert, Button, View, Image, StyleSheet } from 'react-native';
import { Actions } from 'react-native-router-flux';
import { connect } from 'react-redux';
import { Card } from '../common';
import { logoutUser } from '../../actions';

class Menu extends Component {
  constructor(props) {
    super(props);
    
    this.state = {
      username: '',
      password: '',
    };
  }
  
  onLogin() {
    const { username, password } = this.state;

    Alert.alert('Credentials', `${username} + ${password}`);
  }

  render() {
    return (
      <View style={styles.container}>
        <View style={styles.logoContainer}>
          <Image style={styles.logo} source={require('../../../images/similar_logo.png')} />
        </View>
        <Card>
          <View style={styles.buttonContainer}>
            <Button
              style={styles.button}
              title='Vocabulario'
              onPress={() => {
                Actions.vocab();
              }}
            />
          </View> 

          <View style={styles.buttonContainer}>
            <Button
              style={styles.button}
              title='Gramática'
              onPress={() => {
                Actions.grammar();
              }}
            />
          </View>  

          <View style={styles.buttonContainer}>
            <Button style={styles.button} title='Acerca de' onPress={() => {}} />
          </View> 

          <View style={styles.buttonContainer}>
            <Button style={styles.button} title='Cerrar sesión' onPress={this.props.logoutUser} />
          </View> 
        </Card> 

      </View>
    );
  }
}

const styles = StyleSheet.create({
  container: {
    flex: 1,
    alignItems: 'center',
    backgroundColor: '#ccf1f3',
  },
  input: {
    width: 300,
    height: 44,
    padding: 10,
    marginBottom: 10,
  },
  logo: {
    marginTop: 20,
    width: 200,
    height: 173
  },
  button: {
    width: 300,
    alignItems: 'center'
  },
  buttonContainer: { width: '100%', padding: 10 }
});


export default connect(null, { 
  logoutUser
})(Menu);
