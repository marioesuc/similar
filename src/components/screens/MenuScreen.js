// Component that renders screen related to main menu options

import React, { Component } from 'react';
import { Alert, Button, View, Image, StyleSheet, TouchableOpacity, Text } from 'react-native';
import LinearGradient from 'react-native-linear-gradient';
import { Actions } from 'react-native-router-flux';
import { connect } from 'react-redux';
import { Card } from '../common';
import { logoutUser } from '../../actions';
import * as Colors from './styles/Colors';
import Icon from 'react-native-fa-icons';

class Menu extends Component {
  constructor(props) {
    super(props);
    
    this.state = {
      username: '',
      password: '',
    };
  }
  
  onLogin() {
    const { username, password } = this.state;

    Alert.alert('Credentials', `${username} + ${password}`);
  }

  render() {
    return (
      <LinearGradient 
        colors={[Colors.backgroundTopColor, Colors.backgroundBottomColor]}
        style={styles.container}
      >
        <View style={styles.logoContainer}>
          <Image style={styles.logo} source={require('../../../images/similar_logo.png')} />
        </View>
        <View style={styles.buttonsContainer}>
          <TouchableOpacity
            onPress={() => { Actions.vocab(); }}
          >
          <LinearGradient 
          colors={['#658FD6', '#144395']}
          style={styles.signInWithFacebookButton}
        >
              <Icon name='language' style={styles.buttonIcon} />
              <Text style={styles.signInText}>
                  {'Vocabulario'}
              </Text>
            </LinearGradient>
          </TouchableOpacity>

          <TouchableOpacity
            onPress={() => { Actions.speech(); }}
          >
          <LinearGradient 
          colors={['#658FD6', '#144395']}
          style={styles.signInWithFacebookButton}
        >
              <Icon name='comments' style={styles.buttonIcon} />
              <Text style={styles.signInText}>
                  {'Pronunciación'}
              </Text>
            </LinearGradient>
          </TouchableOpacity>

          <TouchableOpacity
            onPress={() => { Actions.grammar(); }}
          >
          <LinearGradient 
          colors={['#658FD6', '#144395']}
          style={styles.signInWithFacebookButton}
        >
              <Icon name='pencil' style={styles.buttonIcon} />
              <Text style={styles.signInText}>
                  {'Gramática'}
              </Text>
            </LinearGradient>
          </TouchableOpacity>

          <TouchableOpacity
            onPress={() => { Actions.about(); }}
          >
          <LinearGradient 
          colors={['#658FD6', '#144395']}
          style={styles.signInWithFacebookButton}
        >
              <Icon name='info' style={styles.buttonIcon} />
              <Text style={styles.signInText}>
                  {'Acerca de'}
              </Text>
            </LinearGradient>
          </TouchableOpacity>

          <TouchableOpacity
            onPress={this.props.logoutUser}
          >
          <LinearGradient 
          colors={['#658FD6', '#144395']}
          style={styles.signInWithFacebookButton}
        >
              <Icon name='sign-out' style={styles.buttonIcon} />
              <Text style={styles.signInText}>
                  {'Cerrar sesión'}
              </Text>
            </LinearGradient>
          </TouchableOpacity>
        </View>
      </LinearGradient>
    );
  }
}

const styles = StyleSheet.create({
  container: {
    flex: 1,
    alignItems: 'center',
    backgroundColor: Colors.appBackground,
  },
  input: {
    width: 300,
    height: 44,
    padding: 10,
    marginBottom: 10,
  },
  logo: {
    marginTop: 20,
    width: 200,
    height: 173
  },
  button: {
    width: 300,
    alignItems: 'center'
  },
  buttonContainer: { width: '100%', padding: 10 },
  signInWithFacebookButton: {
        flexDirection: 'row',
        alignItems: 'center',
        justifyContent: 'center',
        width: 250,
        height: 40,
        padding: 5,
        borderRadius: 5,
        margin: 5,
        elevation: 2
    },
    signInText: {
        color: 'white',
        marginLeft: 5,
        fontFamily: 'HelveticaNeue-Medium',
        fontSize: 17,
        fontWeight: 'bold'
    },
    buttonIcon: {
        marginRight: 5,
        color: 'white',
        fontSize: 25
    },
    buttonsContainer: {
      marginTop: 20,
      width: 280,
      padding: 10,
      alignItems: 'center',
      backgroundColor: 'rgba(255,255,255,0.6)',
      elevation: 2
    }
});


export default connect(null, { 
  logoutUser
})(Menu);
