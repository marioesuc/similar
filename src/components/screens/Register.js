// Component that renders screen related to the register form

import React, { Component } from 'react';
import { Alert, Button, TextInput, View, Image, StyleSheet } from 'react-native';

export default class Register extends Component {
  constructor(props) {
    super(props);
    
    this.state = {
      email: '',
      password: '',
      repeatedPassword: ''
    };
  }
  
  onLogin() {
    const { email, password } = this.state;

    Alert.alert('Credentials', `${email} + ${password}`);
  }

  render() {
    return (
      <View style={styles.container}>
        <View style={styles.logoContainer}>
          <Image style={styles.logo} source={require('../../../images/similar_logo.png')} />
        </View>
        <TextInput
          value={this.state.email}
          onChangeText={(email) => this.setState({ email })}
          placeholder={'E-mail'}
          style={styles.input}
        />
        <TextInput
          value={this.state.password}
          onChangeText={(password) => this.setState({ password })}
          placeholder={'Contraseña'}
          secureTextEntry
          style={styles.input}
        />
        <TextInput
          value={this.state.repeatedPassword}
          onChangeText={(repeatedPassword) => this.setState({ repeatedPassword })}
          placeholder={'Repite contraseña'}
          secureTextEntry
          style={styles.input}
        />

        <Button style={styles.button} title='Registrarse' onPress={() => {}} />
           

      </View>
    );
  }
}

const styles = StyleSheet.create({
  container: {
    flex: 1,
    alignItems: 'center',
    backgroundColor: '#ccf1f3',
  },
  input: {
    width: 300,
    height: 44,
    padding: 10,
    marginBottom: 10,
  },
  logo: {
    marginTop: 20,
    width: 200,
    height: 173
  },
  button: {
    alignItems: 'center'
  }
});
