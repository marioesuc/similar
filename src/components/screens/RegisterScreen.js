// Component that renders screen related to the register form

import React, { Component } from 'react';
import { TextInput, View, Image, Text, StyleSheet, TouchableOpacity } from 'react-native';
import LinearGradient from 'react-native-linear-gradient';
import { connect } from 'react-redux';
import {
  registerEmailChanged,
  registerPasswordChanged,
  registerConfirmPasswordChanged,
  registerUser
} from '../../actions';
import * as Colors from './styles/Colors';

class Register extends Component {

  onEmailChange(text) {
    //registerEmailChanged es la función Action Creator que hemos importado arriba
    this.props.registerEmailChanged(text);
  }

  onPasswordChange(text) {
    this.props.registerPasswordChanged(text);
  }

  onPasswordConfirmChange(text) {
    this.props.registerConfirmPasswordChanged(text);
  }

  onButtonPress() {
    const { email, password, confirmPassword } = this.props;

    this.props.registerUser({ email, password, confirmPassword });
  }


  render() {
    return (
      <LinearGradient 
        colors={[Colors.backgroundTopColor, Colors.backgroundBottomColor]}
        style={styles.container}
      >
        <View style={styles.logoContainer}>
          <Image style={styles.logo} source={require('../../../images/similar_logo.png')} />
        </View>
        <TextInput
          value={this.props.email}
          onChangeText={this.onEmailChange.bind(this)}
          placeholder={'E-mail'}
          style={styles.input}
        />
        <TextInput
          value={this.props.password}
          onChangeText={this.onPasswordChange.bind(this)}
          placeholder={'Introduce una contraseña'}
          secureTextEntry
          style={styles.input}
        />
        <TextInput
          value={this.props.confirmPassword}
          onChangeText={this.onPasswordConfirmChange.bind(this)}
          placeholder={'Repite la contraseña'}
          secureTextEntry
          style={styles.input}
        />

        <TouchableOpacity
          style={styles.buttonContainer}
          onPress={this.onButtonPress.bind(this)}
        >
          <Text style={styles.buttonText}>REGISTRARSE</Text>
        </TouchableOpacity>
           
        <Text style={styles.errorTextStyle}>
          {this.props.error}
        </Text>
      </LinearGradient>
    );
  }
}

const styles = StyleSheet.create({
  container: {
    flex: 1,
    alignItems: 'center',
    backgroundColor: Colors.appBackground,
  },
  input: {
    width: 300,
    height: 44,
    padding: 10,
    marginBottom: 10,
  },
  logo: {
    marginTop: 20,
    width: 200,
    height: 173
  },
  button: {
    alignItems: 'center'
  },
  buttonContainer: {
    backgroundColor: Colors.transparentButton,
    paddingVertical: 15,
    width: 300
  },
  buttonText: {
    textAlign: 'center',
    color: '#FFF',
    fontWeight: '700'
  },
  errorTextStyle: {
    marginTop: 20,
    fontSize: 20,
    alignSelf: 'center',
    color: Colors.errorText
  }
});

const mapStateToProps = ({ register }) => {
  const { email, password, confirmPassword, error } = register;
  return { email, password, confirmPassword, error };
};


export default connect(mapStateToProps, { 
  registerEmailChanged, registerPasswordChanged, registerConfirmPasswordChanged, registerUser
})(Register);
