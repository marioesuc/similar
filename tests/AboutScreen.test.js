// Snapshot testing for the about screen
import 'react-native';
import React from 'react';
import { shallowToJson } from 'enzyme-to-json';
import { shallow } from 'enzyme';

import { About } from '../src/components/screens/AboutScreen';

describe('About Screen Component', () => {
	it('renders correctly', () => {
		const wrapper = shallow(<About />);
		expect(shallowToJson(wrapper)).toMatchSnapshot();
	});
});
