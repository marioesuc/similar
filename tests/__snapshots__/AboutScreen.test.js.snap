// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`About Screen Component renders correctly 1`] = `
<LinearGradient
  colors={
    Array [
      "#C1E8FC",
      "#5D9CBB",
    ]
  }
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "#ccf1f3",
      "flex": 1,
      "padding": 20,
    }
  }
>
  <Component>
    <Image
      source={
        Object {
          "testUri": "../../../images/similar_logo.png",
        }
      }
      style={
        Object {
          "height": 173,
          "width": 200,
        }
      }
    />
  </Component>
  <Component
    style={
      Object {
        "backgroundColor": "white",
        "borderWidth": 0,
        "elevation": 2,
        "margin": 10,
        "padding": 10,
        "width": "95%",
      }
    }
  >
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      style={
        Object {
          "fontSize": 20,
          "fontWeight": "bold",
        }
      }
    >
      ¿Qué es Similar?
    </Text>
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      style={
        Object {
          "padding": 10,
        }
      }
    >
      Similar no es una aplicación cualquiera para aprender idiomas. Nosotros, los creadores, creemos que un buen método de aprendizaje es fundamental para memorizar de la forma más eficiente y rápida posible.
    </Text>
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      style={
        Object {
          "padding": 10,
        }
      }
    >
      Para ello combinamos el uso de las famosas tarjetas de aprendizaje (muy utilizadas en el estudio del idioma Chino) con el reaprovechamiento de los recursos que ofrece el nexo común del latín entre nuestro idioma, el español, y el inglés.
    </Text>
  </Component>
</LinearGradient>
`;
